apps/dashboard/app/new/components/onboarding-content.tsx

```diff
@@ -12,8 +12,23 @@ import { type OnboardingStep, OnboardingWizard } from "./onboarding-wizard";
 export function OnboardingContent() {
   const [currentStepIndex, setCurrentStepIndex] = useState(0);
   const [isConfirmOpen, setIsConfirmOpen] = useState(false);
-  const workspaceStep = useWorkspaceStep();
-  const keyCreationStep = useKeyCreationStep();
+
+  const [stepCount, setStepCount] = useState(0);
+
+  const handleStepChange = (newStepIndex: number) => {
+    setCurrentStepIndex(newStepIndex);
+  };
```

Comment:
const handleStepChange = (newStepIndex: number) => {
  if (newStepIndex >= 0 && newStepIndex < steps.length) {
    setCurrentStepIndex(newStepIndex);
  }
};

Can we do it like this instead of clamping above for boundary checking? This is more explicit and `useEffect`  and `useState` free.